src/main/scala/sbtassembly/Shader.scala

```diff
@@ -81,7 +81,14 @@ private[sbtassembly] object Shader {
     }}
 
     val proc = JJProcessor(jjrules, verbose = level == Level.Debug, true)
-    val files = AssemblyUtils.getMappings(dir, Set())
+
+    /*
+    jarjar MisplacedClassProcessor class transforms byte[] to a class using org.objectweb.asm.ClassReader.getClassName
+    which always translates class names containing '.' into '/', regardless of OS platform.
+    We need to transform any windows file paths in order for jarjar to match them properly and not omit them.
+     */
+    val files = AssemblyUtils.getMappings(dir, Set()).map(f =>
+      if (f._2.contains('\\')) (f._1, f._2.replace('\\', '/')) else f)
```

Comment:
Maybe the added check should be that the shadeRules have changed since the last time you called `assembly`? If you can cache unzip that's usually a good thing.